src/wp-includes/media.php

```diff
@@ -1053,6 +1053,19 @@ function wp_get_attachment_image( $attachment_id, $size = 'thumbnail', $icon = f
 		$html .= ' />';
 	}
 
+	/**
+	 * Filters the markup of image attachments.
+	 *
+	 * @since 5.4.0
+	 *
+	 * @param string       $html       The markup for image attachments.
+	 * @param string[]     $attr       Array of attribute values for the image markup, keyed by attribute name.
+	 * @param WP_Post      $attachment Image attachment post.
+	 * @param string|array $size       Requested size. Image size or array of width and height values
+	 *                                 (in that order). Default 'thumbnail'.
+	 */
+	$html = apply_filters( 'wp_get_attachment_image', $html, $attr, $attachment, $size );
```

Comment:
This also may not be needed if applying the filter later for `the_content` et al.

Reply:
This is needed, as `wp_get_attachment_image` is used for `the_post_thumbnail` and all other places that display image attachments.

Reply:
I don't understand why we need this filter. There is already `wp_get_attachment_image_attributes` which should be everything we need. We should just add this attribute above the filter in `$defaults` - not sure why you removed https://github.com/WordPress/wordpress-develop/pull/123/commits/9693d43acc20703e5eaf68175a155de5f349bef5#diff-67a565933b9d6a2850abd1d7dc5585d5L1016 which I think is a much simpler solution.